pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -99,10 +100,19 @@ func (ic *ifConfigurator) configureContainerLink(
 	mtu int,
 	result *current.Result,
 ) error {
-	// Create HNS Endpoint.
-	endpoint, err := ic.createContainerLink(podName, podNameSpace, containerID, result)
-	if err != nil {
-		return err
+	epName := util.GenerateContainerInterfaceName(podName, podNameSpace)
+	// Search endpoint from local cache.
+	endpoint, found := ic.getEndpoint(epName)
+	if !found {
+		if !ipam.IsInfraContainer(containerNetNS) {
+			return fmt.Errorf("failed to find HNSEndpoint: %s", epName)
+		}
+		// Only create HNS Endpoint for infra container.
+		ep, err := ic.createContainerLink(podName, podNameSpace, containerID, result)
+		if err != nil {
+			return err
+		}
+		endpoint = ep
```

Comment:
It seems duplicated with createContainerLink where there's also createIfNotExists logic.
If it checks endpoint here, it doesn't need to do it again in createContainerLink?

Reply:
Yes, I think I can move the check logic from `createContainerLink`.
Done.